### DISCUSSION TITLE
Discuss these rule differences

### DISCUSSION THREAD

**Discussion by Jacky-LinPeng** [bazel-ios/rules_ios#564] [Q&A] [answered]
Hi, I'm new to Bazel, and I've discovered that Bazel has a lot of build rules, such as: 'bazelbuild/rules_apple', 'bazelbuild/rules_swift', 'bazel-ios/rules_ios',' line/rules_apple_line ', I plan to convert my ios project to bazel, Which rule is recommended? Are they related? I hope to get your reply about my doubts.

**Comment by thiagohmcruz** [comment, marked as answer]
Hello! I'd recommend spending some time reading the `README` file at the root of each repo to understand what they're trying to solve and why. 

If you still have more specific questions after going through that feel free to ask! There's a lot of info and context there so that's why I'd recommend doing that first. 

Also, I'll find many "how we migrated to Bazel" blogs online, this is a good one: https://flare.build/migration-guides/ios-hearts-bazel

That said if you're migrating a small app some of the info in that blog might not directly apply but it's it's a good way to understand key milestones and trade-offs of the rules sets.

I hope this helps!

**Reply by Jacky-LinPeng** [reply]
Thanks a million

https://github.com/bazelbuild/rules_apple/discussions/1499#discussioncomment-3666723

**Comment by jerrymarino** [comment]
I'd ditto what @thiagohmcruz said. This morning I sent over a PR to update the [README](https://github.com/bazel-ios/rules_ios/pull/565/files) - would love any feedback on that 😄 

The other thing I've heard about rules_ios - is that it's used by a lot of the top apps over here. I'm working to build out some social proof / add quotes onto [our website](https://bazel-ios.github.io/)

**Reply by Jacky-LinPeng** [reply]
cc, `bazel-ios/rules_ios`  YYDS
